vkubelet/pod.go

```diff
@@ -179,30 +183,28 @@ func (s *Server) deletePod(ctx context.Context, pod *corev1.Pod) error {
 	addPodAttributes(span, pod)
 
 	var delErr error
-	if delErr = s.provider.DeletePod(ctx, pod); delErr != nil && errors.IsNotFound(delErr) {
+	if delErr = s.provider.DeletePod(ctx, pod); delErr != nil && !errors.IsNotFound(delErr) {
```

Comment:
I don't think providers are returning a K8s error type? I don't think this is working as intended with or without the change.

We need providers to convert to errors that we can understand here.

Reply:
We could use something like `strongerrors.NotFound(err)` in the provider and `strongerrors.IsNotFound(err)` here.

Would be nice to make sure all providers do this, but can fallback to `GetPod` as well.

Reply:
Sounds good

Reply:
https://github.com/virtual-kubelet/virtual-kubelet/pull/403 adds this to providers.

Reply:
btw I think we want this to return when it is a NotFound error because there is nothing to do.

Reply:
Or more specifically, we want to requeue with it's not a NotFound error.

Reply:
we need to hard delete the pod from k8s